console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,197 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +openshift:compatibility-gen:level=1
+
+// ConsolePlugin is an extension for customizing OpenShift web console by
+// dynamically loading code from another service running on the cluster.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+type ConsolePlugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// +kubebuilder:validation:Required
+	Spec ConsolePluginSpec `json:"spec"`
+}
+
+// ConsolePluginSpec is the desired plugin configuration.
+type ConsolePluginSpec struct {
+	// displayName is the display name of the plugin.
+	// The dispalyName should be between 1 and 128 characters.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	DisplayName string `json:"displayName"`
+	// service is a Kubernetes Service that exposes the plugin using a
+	// deployment with an HTTP server. The Service must use HTTPS and
+	// Service serving certificate. The console backend will proxy the
+	// plugins assets from the Service using the service CA bundle.
+	// +kubebuilder:validation:Required
+	Service ConsolePluginService `json:"service"`
+	// proxy is a list of proxies that describe various service type
+	// to which the plugin needs to connect to.
+	// +kubebuilder:validation:Optional
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
+	// i18n is the configuration of plugin's localization resources.
+	// +kubebuilder:validation:Required
+	I18n ConsolePluginI18n `json:"i18n"`
+}
+
+// LoadType is an enumeration of i18n loading types
+// +kubebuilder:validation:Pattern=`^(Preload|Lazy)$`
+type LoadType string
+
+const (
+	// Preload will load all plugin's localization resources during
+	// loading of the plugin.
+	Preload LoadType = "Preload"
+	// Lazy wont preload any plugin's localization resources, instead
+	// will leave thier loading to runtime's lazy-loading.
+	Lazy LoadType = "Lazy"
+)
+
+// ConsolePluginI18n holds information on localization resources that are served by
+// the dynamic plugin.
+type ConsolePluginI18n struct {
+	// load indicates how the plugin's localization resource should be loaded.
+	// +kubebuilder:validation:Required
+	Load LoadType `json:"load"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// backend provides information about endpoint to which the request is proxied to.
```

Comment:
The name of this field and the comment don't match, do you want to call it endpoint rather than backend?